orangecontrib/text/widgets/owscoredocuments.py

```diff
@@ -116,22 +115,31 @@ def _preprocess_words(
     with words preprocessors that change words (e.g. normalization) must
     be applied to words too.
     """
+    # workaround to preprocess words
+    # TODO: currently preprocessors work only on corpus, when there will be more
+    #  cases like this think about implementation of preprocessors for a list
+    #  of strings
+    words_feature = StringVariable("words")
+    words_c = Corpus(
+        Domain([], metas=[words_feature]),
+        metas=np.array([[w] for w in words]),
+        text_features=[words_feature]
+    )
     # only transformers and normalizers preprocess on the word level
     pps = [
         pp
         for pp in corpus.used_preprocessor.preprocessors
         if isinstance(pp, (BaseTransformer, BaseNormalizer))
     ]
     for i, pp in enumerate(pps):
-        # TODO: _preprocess is protected make it public
-        words = [pp._preprocess(w) for w in words]
+        words_c = pp(words_c)
         callback((i + 1) / len(pps))
-    return words
+    return [w[0] for w in words_c.tokens if len(w) > 0]
```

Comment:
This may result in an empty list, which is could probably cause some unexpected behaviour.

Reply:
Should be ok now.